lib/aptible/cli/helpers/security_key.rb

```diff
@@ -0,0 +1,136 @@
+module Aptible
+  module CLI
+    module Helpers
+      module SecurityKey
+        U2F_LOGGER = Logger.new(
+          ENV['U2F_DEBUG'] ? STDERR : File.open(File::NULL, 'w')
+        )
+
+        class AuthenticatorParameters
+          attr_reader :origin, :challenge, :app_id, :version, :key_handle
+          attr_reader :request
+
+          def initialize(origin, challenge, app_id, device)
+            @origin = origin
+            @challenge = challenge
+            @app_id = app_id
+            @version = device.version
+            @key_handle = device.key_handle
+
+            @request = {
+              'challenge' => challenge,
+              'appId' => app_id,
+              'version' => version,
+              'keyHandle' => key_handle
+            }
+          end
+        end
+
+        class ThrottledAuthenticator
+          attr_reader :pid
+
+          def initialize(auth, pid)
+            @auth = auth
+            @pid = pid
+          end
+
+          def exited(_status)
+            [Authenticator.spawn(@auth), nil]
+          end
+
+          def self.spawn(auth)
+            pid = Process.spawn(
+              'sleep', '2',
+              in: :close, out: :close, err: :close,
+              close_others: true
+            )
+
+            U2F_LOGGER.debug("#{self} #{auth.key_handle}: spawned #{pid}")
+
+            new(auth, pid)
+          end
+        end
+
+        class Authenticator
+          attr_reader :pid
+
+          def initialize(auth, pid, out_read, err_read)
+            @auth = auth
+            @pid = pid
+            @out_read = out_read
+            @err_read = err_read
+          end
+
+          def exited(status)
+            out, err = [@out_read, @err_read].map(&:read).map(&:chomp)
+
+            if status.exitstatus == 0
+              U2F_LOGGER.info("#{self.class} #{@auth.key_handle}: ok: #{out}")
+              [nil, JSON.parse(out)]
+            else
+              U2F_LOGGER.warn("#{self.class} #{@auth.key_handle}: err: #{err}")
+              [ThrottledAuthenticator.spawn(@auth), nil]
+            end
+          ensure
+            [@out_read, @err_read].each(&:close)
+          end
+
+          def self.spawn(auth)
+            in_read, in_write = IO.pipe
+            out_read, out_write = IO.pipe
+            err_read, err_write = IO.pipe
+
+            pid = Process.spawn(
+              'u2f-host', '-aauthenticate', '-o', auth.origin,
```

Comment:
Would it make sense to first check if `u2f-host` is in the path or not, so that we avoid `Errno::ENOENT: No such file or directory - u2f-host` or whatever the traceback would be when it's not present? (eg on windows, or people that install the gem directly)?